### DISCUSSION TITLE
posts/python-coroutine-programming-2/

### DISCUSSION THREAD

**Discussion by giscus** [humminglab/blog.humminglab.io#1] [Announcements]
# posts/python-coroutine-programming-2/

이글은 Python 비동기 프로그래밍 제대로 이해하기(1/2) 에 이어서 작성한 글이다.
Asyncio Python 3.4 에서는 그동안 Twisted, Tornado와 같이 별도의 library로 제공되던 event loop 방식의 비동기 프로그래밍이 asyncio (PEP 3156 – Asynchronous IO Support Rebooted: the “asyncio” Module) 표준 라이브러리로 새로 추가되었다.
각각의 event loop 구현이 비슷하지만 약간의 차이가 있어서 이들을 혼용하여 사용할 때 차이점을 이해하는데 부담이 있지만, 시간이 지나며 이들도 asyncio로 통합 지원하는 방향으로 되는 것 같다. 참고로 2018년 3월에 새로 릴리즈된 Tornado 5.

https://blog.humminglab.io/posts/python-coroutine-programming-2/#async-internal

**Comment by L0Z1K** [comment]
너무 잘 읽었습니다. 이 글 덕분에 제대로 개념을 잘 잡은 것 같습니다. 읽다가 오타를 조금 발견해서 PR을 올려두었으니 확인해주셔도 좋을 것 같습니다. 

https://github.com/humminglab/blog.humminglab.io/pull/11

**Reply by aqwerf** [reply]
잘못된 부분 수정해 주셔서 고맙습니다.

Tornado 로 비동기 함수를 사용하여 API 서버를 만들면서 제대로 이해 하려고 정리하면서 쓴글인데, 
다시 보아도 python async 는 히스토리가 복잡해서 이해하기가 쉽지 않네요. ^^

**Reply by L0Z1K** [reply]
ㅎㅎ 감사합니다!
